jerry-core/rcs/rcs-records.h

```diff
@@ -0,0 +1,169 @@
+ *
+ * Layout:
+ * ------- header -----------------------
+ * type (4 bits)
+ * magic string id  (12 bits)
+ * pointer to prev (16 bits)
+ * --------------------------------------
+ */
+#define RCS_MAGIC_STR_HEADER_SIZE                RCS_DYN_STORAGE_LENGTH_UNIT
+
+#define RCS_MAGIC_STR_HEADER_ID_POS              RCS_HEADER_FIELD_BEGIN_POS
+#define RCS_MAGIC_STR_HEADER_ID_WIDTH            12u
+
+#define RCS_MAGIC_STR_HEADER_PREV_POS            (RCS_MAGIC_STR_HEADER_ID_POS + RCS_MAGIC_STR_HEADER_ID_WIDTH)
+
+/**
+ * Free record
+ * Doesn't hold any data.
+ *
+ * Layout:
+ * ------- header -----------------------
+ * type (4 bits)
+ * length (12 bits)
+ * pointer to prev (16 bits)
+ * --------------------------------------
+ */
+#define RCS_FREE_HEADER_SIZE                     RCS_DYN_STORAGE_LENGTH_UNIT
+
+#define RCS_FREE_HEADER_LENGTH_POS               RCS_HEADER_FIELD_BEGIN_POS
+#define RCS_FREE_HEADER_LENGTH_WIDTH             (14u - RCS_DYN_STORAGE_LENGTH_UNIT_LOG)
+
+#define RCS_FREE_HEADER_PREV_POS                 (RCS_FREE_HEADER_LENGTH_POS + RCS_FREE_HEADER_LENGTH_WIDTH)
+
+/*
+ *  Setters
+ */
+extern void rcs_record_set_type (rcs_record_t *, rcs_record_type_t);
+extern void rcs_record_set_prev (rcs_record_set_t *, rcs_record_t *, rcs_record_t *);
+extern void rcs_record_set_size (rcs_record_t *, size_t);
+extern void rcs_record_set_alignment_bytes_count (rcs_record_t *, size_t);
+extern void rcs_record_set_hash (rcs_record_t *, lit_string_hash_t);
+extern void rcs_record_set_charset (rcs_record_set_t *, rcs_record_t *, const lit_utf8_byte_t *, lit_utf8_size_t);
+extern void rcs_record_set_magic_str_id (rcs_record_t *, lit_magic_string_id_t);
+extern void rcs_record_set_magic_str_ex_id (rcs_record_t *, lit_magic_string_ex_id_t);
+
+/*
+ *  Getters
+ */
+extern rcs_record_type_t rcs_record_get_type (rcs_record_t *);
+extern rcs_record_t *rcs_record_get_prev (rcs_record_set_t *, rcs_record_t *);
+extern size_t rcs_record_get_size (rcs_record_t *);
+extern size_t rcs_header_get_size (rcs_record_t *);
+extern size_t rcs_record_get_alignment_bytes_count (rcs_record_t *);
+extern lit_string_hash_t rcs_record_get_hash (rcs_record_t *);
+extern lit_utf8_size_t rcs_record_get_length (rcs_record_t *);
+extern lit_utf8_size_t rcs_record_get_charset (rcs_record_set_t *, rcs_record_t *, const lit_utf8_byte_t *, size_t);
+extern lit_magic_string_id_t rcs_record_get_magic_str_id (rcs_record_t *);
+extern lit_magic_string_ex_id_t rcs_record_get_magic_str_ex_id (rcs_record_t *);
+extern ecma_number_t rcs_record_get_number (rcs_record_set_t *, rcs_record_t *);
+
+extern rcs_record_t *rcs_record_get_first (rcs_record_set_t *);
+extern rcs_record_t *rcs_record_get_next (rcs_record_set_t *, rcs_record_t *);
+
+extern bool rcs_record_is_equal (rcs_record_set_t *, rcs_record_t *, rcs_record_t *);
+extern bool rcs_record_is_equal_utf8 (rcs_record_set_t *, rcs_record_t *, const lit_utf8_byte_t *, lit_utf8_size_t);
+
+#endif
```

Comment:
Usually, we put comments after `#endif` to describe condition, corresponding to code just above the `#endif`. For example, `!RCS_RECORDS_H` in the case.